apps/api/src/app/controllers/__tests__/queryACLs.e2e.spec.ts

```diff
@@ -0,0 +1,351 @@
+                            it('should return not found', async () => {
+                                const resourceToFind = confidentialResources[0];
+
+                                const detailEndpoint = buildDetailEndpoint(resourceToFind.id);
+
+                                await request(app.getHttpServer())
+                                    .get(detailEndpoint)
+                                    .expect(httpStatusCodes.notFound);
+                            });
+                        });
+                    });
+                });
+            });
+
+            (
+                [
+                    [CoscradUserRole.superAdmin, 'when the user is a COSCRAD admin'],
+                    [CoscradUserRole.projectAdmin, 'when the user is a project admin'],
+                ] as const
+            ).forEach(([userRole, description]) => {
+                describe(description, () => {
+                    const dummyAdminUser = dummyUser.clone({
+                        roles: [userRole],
+                    });
+
+                    const dummyAdminUserWithGroups = new CoscradUserWithGroups(dummyAdminUser, []);
+
+                    beforeAll(async () => {
+                        ({ app, testRepositoryProvider } = await setUpIntegrationTest(
+                            {
+                                ARANGO_DB_NAME: generateRandomTestDatabaseName(),
+                                BASE_DIGITAL_ASSET_URL,
+                            },
+                            {
+                                testUserWithGroups: dummyAdminUserWithGroups,
+                            }
+                        ));
+                    });
+
+                    beforeEach(async () => {
+                        await testRepositoryProvider.testSetup();
+
+                        await testRepositoryProvider.addFullSnapshot(
+                            buildInMemorySnapshot({
+                                users: [dummyAdminUser],
+                                userGroups: [dummyGroup],
+                                resources: {
+                                    [resourceType]: allResourcesOfCurrentType,
+                                },
+                            })
+                        );
+                    });
+
+                    afterEach(async () => {
+                        // await testRepositoryProvider.testTeardown();
+                        // await arangoConnectionProvider.dropDatabaseIfExists();
+                        // await app.close();
+                    });
+
+                    describe('when querying for many resources (fetch many)', () => {
+                        it('should find all resources of the given kind', async () => {
+                            const res = await request(app.getHttpServer()).get(indexEndpoint);
+
+                            assertOkResponse(res, allResourcesOfCurrentType);
```

Comment:
Is the resources we expect to find and the user the only thing that changes? Maybe we can have one big loop over all 4 cases?